dump/main.py

```diff
@@ -0,0 +1,214 @@
+import os,sys,json,re
+import click
+from tabulate import tabulate
+
+sys.path.append(os.path.dirname(__file__))
+import plugins
+from dump.match_infra import RedisSource, JsonSource
+from sonic_py_common import multi_asic
+from utilities_common.constants import DEFAULT_NAMESPACE
+
+# Autocompletion Helper
+def get_available_modules(ctx, args, incomplete):
+    return [k for k in plugins.dump_modules.keys() if incomplete in k]
+
+# Display Modules Callback
+def show_modules(ctx, param, value):
+    if not value or ctx.resilient_parsing:
+        return
+    header = ["Module", "Identifier"]
+    display = []
+    for mod in plugins.dump_modules:
+        display.append((mod, plugins.dump_modules[mod].ARG_NAME))
+    click.echo(tabulate(display, header))
+    ctx.exit()
+             
+@click.group()
+def dump():
+    pass
+
+@dump.command()
+@click.pass_context
+@click.argument('module', required=True, type=str, autocompletion=get_available_modules)
+@click.argument('identifier', required=True, type=str) 
+@click.option('--show', '-s', is_flag=True, default=False, help='Display Modules Available', is_eager=True, expose_value=False, callback=show_modules)
+@click.option('--db', '-d', multiple=True, help='Only dump from these Databases or the CONFIG_FILE')
+@click.option('--table', '-t', is_flag=True, default=False, help='Print in tabular format', show_default=True)
+@click.option('--key-map', '-k', is_flag=True, default=False, help="Only fetch the keys matched, don't extract field-value dumps", show_default=True)
+@click.option('--verbose', '-v', is_flag=True, default=False, help="Prints any intermediate output to stdout useful for dev & troubleshooting", show_default=True)
+@click.option('--namespace', '-n', default=DEFAULT_NAMESPACE, type=str, show_default=True, help='Dump the redis-state for this namespace.')  
+def state(ctx, module, identifier, db, table, key_map, verbose, namespace):
+    """
+    Dump the current state of the identifier for the specified module from Redis DB or CONFIG_FILE
+    """
+    if not multi_asic.is_multi_asic() and namespace != DEFAULT_NAMESPACE:
+        click.echo("Namespace option is not valid for a single-ASIC device")
+        ctx.exit()
+    
+    if multi_asic.is_multi_asic() and (namespace != DEFAULT_NAMESPACE and namespace not in multi_asic.get_namespace_list()):
```

Comment:
With the recent changes in https://github.com/Azure/sonic-utilities/pull/1712/files ; load_db_config will have to be done before invoking multi_asic.get_namespace_list

Reply:
Added

Reply:
my mistake, load_db_config() is not require here as get_namespace_list does not connect to DB to get the namespaces list. This is not required here.

Reply:
Noted and removed